src/core/gosling-track-model.ts

```diff
@@ -334,15 +341,15 @@ export class GoslingTrackModel {
             return 1;
         }
 
-        const vSpec = spec.visibility;
+        const vSpec = spec.visibility as SizeVisibilityCondition;
```

Comment:
As we only allow `SizeVisibilityCondition` when `target: mark`, how about reflecting this in `gosling.schema.ts`?

So, the updated schema will be like:

```js
interface CommonVisibilityCondition {
    operation: LogicalOperation;
}
export interface SizeVisibilityCondition extends CommonVisibilityCondition {
    target: 'track' | 'mark';
    measure: 'width' | 'height';
    threshold: number | '|xe-x|';
    conditionPadding?: number;
    transitionPadding?: number;
}

export interface ZoomLevelVisibilityCondition extends CommonVisibilityCondition {
    target: 'track';
    measure: 'zoomLevel';
    threshold: number;
}
```
This would make the schema more precise.